site/content/docs/tutorials/hf-gcs-transfer/index.md

```diff
@@ -0,0 +1,429 @@
+ 
+```bash
+gcloud container clusters create-auto ${CLUSTER_NAME} \
+  --project=${PROJECT_ID} \
+  --region=${REGION} \
+  --labels=created-by=ai-on-gke,guide=hf-gcs-transfer 
+```
+
+
+### Create a Kubernetes secret for Hugging Face credentials
+
+In your shell session, do the following:
+
+1. Configure `kubectl` to communicate with your cluster:
+
+    ```bash
+    gcloud container clusters get-credentials ${CLUSTER_NAME} --location=${REGION}
+    ```
+
+1. Create a Kubernetes Secret that contains your Hugging Face token. This is only required for [gated models](https://huggingface.co/docs/hub/models-gated):
+
+    ```bash
+    kubectl create secret generic hf-secret \
+      --from-literal=hf_api_token=${HF_TOKEN} \
+      --dry-run=client -o yaml | kubectl apply -f -
+    ```
+
+
+## Create your Cloud Storage bucket
+
+Now, create the Cloud Storage bucket for the model weights, by running the following command.
+> Note: Cloud Storage bucket's names must be globally unique, and you must have the Storage Admin (`roles/storage.admin`) IAM role for the project where the bucket is created. See [Create a Bucket](https://cloud.google.com/storage/docs/creating-buckets) for details.
+
+```bash
+export BUCKET_NAME=${PROJECT_ID}-meta-llama-3-8b
+export BUCKET_URI=gs://${BUCKET_NAME}
+gcloud storage buckets create ${BUCKET_URI} --project=${PROJECT_ID}
+```
+
+## Deploy the Kubernetes Job to populate the Cloud Storage Bucket
+
+1. Configure access for the `producer-job` Job, to the Cloud Storage bucket. 
+
+    To make your Cloud Storage bucket accessible by your GKE cluster, authenticate using Workload Identity Federation for GKE with the Cloud Storage bucket. 
+    > Note: if you don't have Workload Identity Federation for GKE enabled, follow [these steps](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#enable_on_clusters_and_node_pools) to enable it.
+
+    Grant the Storage Admin (`roles/storage.admin`) IAM role for Cloud Storage to the Kubernetes ServiceAccount by running the following commands. If you are using a custom workload identity pool, you will need to update the workload identity pool name in the command below. Custom Workload Identitiy pools are not supported in Autopilot clusters.
+
+    ```bash
+    export PROJECT_NUMBER=$(gcloud projects describe ${PROJECT_ID} --format="value(projectNumber)")
+    export WORKLOAD_IDENTITY_POOL=${PROJECT_ID}.svc.id.goog
+    export NAMESPACE=default
+    export SERVICE_ACCOUNT=hf-sa
+    gcloud storage buckets add-iam-policy-binding ${BUCKET_URI} \
+    --member "principal://iam.googleapis.com/projects/${PROJECT_NUMBER}/locations/global/workloadIdentityPools/${WORKLOAD_IDENTITY_POOL}/subject/ns/${NAMESPACE}/sa/${SERVICE_ACCOUNT}" \
+    --role "roles/storage.admin	
+    ```
+
+1. Deploy the Job and Kubernetes Service Account below.
+
+    ```bash
+    kubectl apply -f - <<EOF
```

Comment:
I need to fix a couple of the ENV variables for the producer job and the inference server, because  they don't get set properly when I use kubectl apply -f - <<EOF. I will fix this tomorrow.

Reply:
I fixed this.